src/pipes/sort.pipe.ts

```diff
@@ -0,0 +1,22 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'sortBy'
+})
+export class SortPipe implements PipeTransform {
+
+  transform(array: Array<any>, prop: string, key: string): Array<any> {
+    const toLowerCase = str => String(str).toLowerCase();
+    if (array && array.length) {
+      let sortFn;
+      switch (key) {
```

Comment:
I would advise moving this logic by defining the sorting function to some function: getSortingMethod